src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -864,39 +864,39 @@ private function getHeaderValue(?array $headerDic, string $key, string $type, $d
      */
     protected function getXrefData(string $pdfData, int $offset = 0, array $xref = []): array
     {
-        $startxrefPreg = preg_match(
-            '/[\r\n]startxref[\s]*[\r\n]+([0-9]+)[\s]*[\r\n]+%%EOF/i',
+        // If the $offset is currently pointed at whitespace, bump it
+        // forward until it isn't; affects loosely targetted offsets
+        // for the 'xref' keyword
+        // See: https://github.com/smalot/pdfparser/issues/673
+        $bumpOffset = $offset;
+        while (preg_match('/\s/', substr($pdfData, $bumpOffset, 1))) {
+            ++$bumpOffset;
+        }
+
+        // Find all startxref tables from this $offset forward
+        $startxrefPreg = preg_match_all(
+            '/(?<=[\r\n])startxref[\s]*[\r\n]+([0-9]+)[\s]*[\r\n]+%%EOF/i',
             $pdfData,
-            $matches,
-            \PREG_OFFSET_CAPTURE,
+            $startxrefMatches,
+            \PREG_SET_ORDER,
             $offset
         );
```

Comment:
Using `$startxrefMatches` as third parameter and to store result of `preg_match_all` looks weird. It seems to work, but I am wondering which one of these two is stored. Based on your code you assume its the latter. But you also use it as it contains the matches (e.g. line 890).

I would like to use two separate variables here to improve code readability.

Reply:
`$startxrefPreg` is the result of the `preg_match_all()` and `$startxrefMatches` stores the associated matches. So they are already two different variables? Unless I'm reading your comment incorrectly...

Reply:
My bad, all good.